File: backend/app/routers/courses.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import List, Optional
from ..db import get_db
from .. import models, schemas

router = APIRouter(prefix="/courses", tags=["courses"])

@router.post("", response_model=schemas.CourseOut)
def create_course(payload: schemas.CourseCreate, db: Session = Depends(get_db)):
    # Moniteur obligatoire + doit exister
    if not db.get(models.Monitor, payload.monitor_id):
        raise HTTPException(404, "Monitor not found")
    c = models.Course(
        name=payload.name,
        description=payload.description,
        monitor_id=payload.monitor_id
    )
    db.add(c); db.commit(); db.refresh(c)
    return c

@router.get("", response_model=List[schemas.CourseOut])
def list_courses(monitor_id: Optional[int] = None, db: Session = Depends(get_db)):
    q = db.query(models.Course)
    if monitor_id:
        q = q.filter(models.Course.monitor_id == monitor_id)
    return q.order_by(models.Course.id.desc()).all()

File: frontend/src/api/courses.ts
import { api } from "../api/client";

export type Course = {
  id: number;
  name: string;
  description?: string | null;
  monitor_id?: number | null;
};

export async function listCourses(monitor_id?: number): Promise<Course[]> {
  const { data } = await api.get("/courses", { params: monitor_id ? { monitor_id } : {} });
  return data;
}

export async function createCourse(payload: { name: string; description?: string; monitor_id?: number }) {
  const { data } = await api.post("/courses", payload);
  return data as Course;
}

export async function getOrCreateSessionByDate(courseId: number, date: string) {
  const { data } = await api.post(`/courses/${courseId}/sessions/by-date`, { date });
  return data as { id:number; course_id:number; index:number; date:string|null };
}
